docs/tutorials/control/flutter-app.md

```diff
@@ -0,0 +1,732 @@
+
+In the VS Code file explorer, right click <file>lib/</file> and click **New File** - let’s name this new file <file>home_screen.dart</file>.
+
+Paste the contents of this code block into that new file:
+
+```{class="line-numbers linkable-line-numbers"}
+import 'package:flutter/material.dart';
+import 'package:flutter_dotenv/flutter_dotenv.dart';
+import 'package:viam_sdk/protos/app/app.dart';
+import 'package:viam_sdk/viam_sdk.dart';
+
+class HomeScreen extends StatefulWidget {
+  const HomeScreen({super.key});
+
+  @override
+  State<HomeScreen> createState() => _HomeScreenState();
+}
+
+class _HomeScreenState extends State<HomeScreen> {
+  late Viam _viam;
+  late Organization _organization;
+  List<Location> _locations = [];
+  bool _loading = true;
+
+  @override
+  void initState() {
+    _getData();
+    super.initState();
+  }
+
+  void _getData() async {
+    try {
+      _viam = await Viam.withApiKey(dotenv.env['API_KEY_ID']?? '', dotenv.env['API_KEY']?? '');
+      _organization = (await _viam.appClient.listOrganizations()).first;
+      _locations = await _viam.appClient.listLocations(_organization);
+
+      // in Flutter setState tells the UI to rebuild the widgets whose state has changed,
+      // this is how we change from showing a loading screen to a list of values
+      setState(() {
+        _loading = false;
+      });
+    } catch (e) {
+      print(e);
+    }
+  }
+
+  /// This method will navigate to a specific [Location].
+  void _navigateToLocation(Location location) {
+    // Navigator.of(context)
+    //     .push(MaterialPageRoute(builder: (_) => LocationScreen(_viam, location)));
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(title: const Text('Locations')),
+      // If we are loading, show a loading indicator.
+      // Otherwise, show a list of [Locations]s.
+      body: _loading
+          ? Center(
+              child: const CircularProgressIndicator.adaptive(),
+            )
+          : // Build a list from the [_locations] state.
+          ListView.builder(
+              itemCount: _locations.length,
+              itemBuilder: (_, index) {
+                final location = _locations[index];
+                return ListTile(
+                  title: Text(location.name),
+                  onTap: () => _navigateToLocation(location),
+                  trailing: const Icon(Icons.chevron_right),
+                );
+              },
+            ),
+    );
+  }
+}
+```
+
+### Get the Viam API key
+
+Notice in the file the following line:
+
+```{class="line-numbers linkable-line-numbers"}
+_viam = await Viam.withApiKey(dotenv.env['API_KEY_ID']?? '', dotenv.env['API_KEY']?? '');
+```
+
+You need to get the API keys from the Viam app and store them in an environmental variables file to so your mobile app can access your smart machine:
```

Comment:
Nit: While the API key indeed has two parts, we still refer to it in the singular. Plural is for when I rotate between multiple keys, or have multiple keys with different perms, or whatever.

Also, here's an opportunity for some light "what is a key" clarification. Maybe something like:

```suggestion
This line in the code defines how your Flutter app authenticates to the Viam platform, by referencing two environment variables that together comprise your Viam API key.
Follow the steps below to get your API key and create an environmental variables file to store them in:
```

Reply:
```suggestion
This line in the code defines how your Flutter app authenticates to the Viam platform, by referencing two environment variables that together comprise your Viam API key.

Follow the steps below to get your API key and create an environment variables file to store them in:
```
Great suggestion. Just going to change "environmental" to "environment" in the second instance since that seems more common?